nesc/whip6/platforms/boards/cc26xxbased/VIMSOverridePub.nc

```diff
@@ -0,0 +1,36 @@
+/*
+ * whip6: Warsaw High-performance IPv6.
+ *
+ * Copyright (c) University of Warsaw
+ * All rights reserved.
+ *
+ * This file is distributed under the terms in the attached LICENSE
+ * files.
+ *
+ */
+
+
+#include <driverlib/vims.h>
+
+#define CC26XX_PLACE_IN_GPRAM __attribute__((section(".gpram")))
+
```

Comment:
What happens to a variable annotated with this attribute/define when an app is compiled without `CC26XX_VIMS_GPRAM_MODE`?

Reply:
Possibly some hard fault on write to array, maybe even at declaration. Symbols declared with this attribute hold reference to GPRAM no matter if it was compiled with or without `CC26XX_VIMS_GPRAM_MODE=1`.

It would be nice if we could somehow throw error if we use `CC26XX_PLACE_IN_GPRAM` and if `CC26XX_VIMS_GPRAM_MODE=0` or is not defined. Don't know how to do this on `CC26XX_PLACE_IN_GPRAM` usage with C preprocessor though.

Reply:
Proponuję coś w tym stylu:

`if ((! defined(CC26XX_VIMS_GPRAM_MODE)) || ((CC26XX_VIMS_GPRAM_MODE) != 1))

#define CC26XX_PLACE_IN_GPRAM @Cannot_place_objects_in_CC26XX_GPRAM_without_CC26XX_VIMS_GPRAM_MODE_set_to_one

#else

#define CC26XX_PLACE_IN_GPRAM __attribute__((section(".gpram")))

#endif`

Oczywiście przetestuj, czy działa.

Reply:
Hmm... Some conditional defining `CC26XX_PLACE_IN_GPRAM` (depending on whether `CC26XX_VIMS_GPRAM_MODE` is defined) won't work, right?